### PR TITLE
Db helpers

### PR DESCRIPTION
**Pull request by mikolajgrzyb**
In this PR I've created layer for more complicated interactions with Repo. Please comment! 
@Xypus, @marcinwyszynski , @pjar

### CODE REVIEW

**Comment by marcinwyszynski** [comment]
Please add the `.hex` directory to the `.gitignore` file.

File: mix.lock [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-  "combine": {:hex, :combine, "0.8.0"},
-  "connection": {:hex, :connection, "1.0.2"},
-  "cowboy": {:hex, :cowboy, "1.0.4"},
-  "cowlib": {:hex, :cowlib, "1.0.2"},
-  "db_connection": {:hex, :db_connection, "0.2.5"},
-  "decimal": {:hex, :decimal, "1.1.2"},
-  "ecto": {:hex, :ecto, "1.1.8"},
-  "espec": {:hex, :espec, "0.8.21"},
-  "espec_phoenix": {:hex, :espec_phoenix, "0.2.1"},
+%{"certifi": {:hex, :certifi, "0.4.0", "a7966efb868b179023618d29a407548f70c52466bf1849b9e8ebd0e34b7ea11f", [:rebar3], []},
```
Comment (marcinwyszynski) [inline mix.lock:1]: What happened here? You vendorized dependencies?

File: spec/services/db_helpers_spec.exs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  let :changeset, do: User.registration_changeset(%User{}, %{
+    first_name: "Mark",
+    last_name: "test", 
+    google_id: "test1234", 
+    email: "test@test.com"
+  })
+
+
+  describe ".find_or_create" do 
+    context "when user is not in the database" do
```
Comment (marcinwyszynski) [inline spec/services/db_helpers_spec.exs:16]: 🍪

File: spec/services/db_helpers_spec.exs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      end 
+
+      it "does not change count of objects" do 
+        subject
+        query = from(p in User, select: count(p.id))
+        expect(Repo.one(query)).to eq 1
+      end 
+    end 
+  end 
+end
```
Comment (marcinwyszynski) [inline spec/services/db_helpers_spec.exs:46]: It's a good practice to end every file with a newline.

File: web/controllers/auth_controller.ex [hunk 4]
```diff
@@ -1,17 +1,18 @@
 defmodule Janitor.AuthController do
+  import Janitor.DBHelpers
   use Janitor.Web, :controller
   use Timex
   alias Janitor.User
-
+
```
Comment (marcinwyszynski) [inline web/controllers/auth_controller.ex:6]: You can set up your editor to clean up whitespace in empty lines. It's a good practice to do so.

File: web/services/db_helpers.ex [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    get_by_map = Map.take(changeset.params, [param])
+    entity = Repo.get_by(structName, get_by_map)
+    case entity do 
+      nil -> create(changeset)
+      _ -> entity    
+    end
+  end 
+
+  defp create(changeset) do 
+    {:ok, entity} = Repo.insert(changeset)
```
Comment (marcinwyszynski) [inline web/services/db_helpers.ex:14]: Y U NO check for errors? What's the use of this method anyway? It obscures more than it helps.

Comment (mikolajgrzyb) [inline web/services/db_helpers.ex:14]: whoops...

**Pull request merged**